compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1616,19 +1614,29 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                         | ObligationCauseCode::ObjectCastObligation(..)
                         | ObligationCauseCode::OpaqueType
                 );
-                if let Err(new_err) = self.at(&obligation.cause, obligation.param_env).eq_exp(
+                let expected_ty = data.term.ty().unwrap();
+                if let Err(new_err) = ocx.eq_exp(
```

Comment:
```suggestion
                let values = if let Err(new_err) = ocx.eq_exp(
```
i think initiatlizing values directly here might be nicer